apps/api/v2/src/ee/event-types/event-types_2024_06_14/services/private-links-output.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from "@nestjs/common";
+
+import { type PrivateLinkData } from "@calcom/platform-libraries/private-links";
+import {
+  PrivateLinkOutput_2024_06_14,
+  TimeBasedPrivateLinkOutput_2024_06_14,
+  UsageBasedPrivateLinkOutput_2024_06_14,
+} from "@calcom/platform-types";
+
+@Injectable()
+export class PrivateLinksOutputService_2024_06_14 {
+  constructor() {}
+
+  /**
+   * Transform platform library internal data to API v2 output
+   */
+  transformToOutput(data: PrivateLinkData): PrivateLinkOutput_2024_06_14 {
+    const baseData = {
+      linkId: data.id.toString(),
+      link: data.link,
+      eventTypeId: data.eventTypeId,
+      isExpired: data.isExpired,
+      bookingUrl: data.bookingUrl,
+    };
+
+    // Return time-based link if expiresAt is set
+    if (data.expiresAt !== null && data.expiresAt !== undefined) {
+      return {
+        ...baseData,
+        expiresAt: data.expiresAt,
+      } as TimeBasedPrivateLinkOutput_2024_06_14;
```

Comment:
We can replace typecasting here and in other places using `plainToClass` - see example in `apps/api/v2/src/modules/organizations/delegation-credentials/organizations-delegation-credential.controller.ts` or other files.

`plainToClass` accepts as first argument the output DTO you have created, then the data, and then optional properties.